platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java

```diff
@@ -53,21 +56,38 @@ public static boolean isLocal(@NonNull final Address address) {
     }
 
     /**
-     * Get the number of addresses currently in the address book that are running on this computer. When the browser is
+     * Get the number of roster entries currently in the roster that are running on this computer. When the browser is
      * run with a config.txt file, it can launch multiple copies of the app simultaneously, each with its own TCP/IP
      * port. This method returns how many there are.
      *
-     * @param addressBook the address book to check
+     * @param roster the roster to check
      * @return the number of local addresses
      */
-    public static int getLocalAddressCount(@NonNull final AddressBook addressBook) {
-        Objects.requireNonNull(addressBook, "The addressBook must not be null.");
-        int count = 0;
-        for (final Address address : addressBook) {
-            if (isLocal(address)) {
-                count++;
-            }
+    public static int getLocalAddressCount(@NonNull final Roster roster) {
+        Objects.requireNonNull(roster, "The roster must not be null.");
+        return (int) roster.rosterEntries().stream()
+                .filter(AddressBookNetworkUtils::isLocal)
+                .count();
+    }
+
+    /**
+     * Check if the rosterEntry is local to the machine.
+     *
+     * @param rosterEntry the rosterEntry to check
+     * @return true if the rosterEntry is local to the machine, false otherwise
+     * @throws IllegalStateException if the locality of the rosterEntry cannot be determined.
+     */
+    public static boolean isLocal(@NonNull final RosterEntry rosterEntry) {
+        Objects.requireNonNull(rosterEntry, "The rosterEntry must not be null.");
+        try {
+            final String internalHostname = RosterUtils.fetchHostname(rosterEntry, 1);
```

Comment:
The current logic in the `buildRoster()` would skip a duplicate service endpoint if both internal and external endpoints in the AddressBook are the same, which is true for the majority of the nodes on the network in mainnet.

Further, the Roster specification doesn't mandate having more than a single endpoint in the list.

So fetching an entry at index 1 unconditionally may and likely will fail.

Additionally, the current logic in the Consensus module would only ever use the entry at index 0 to establish network connections.

With all this in mind, you probably want to run this check against the entry at index 0. Or perhaps remove it altogether, with a note in the "node configuration/endpoint override" story that it needs to be restored when those overrides are implemented.

Reply:
Nope.   It has to use the internal ip address until we have an alternate means of determining which node is starting. 

Issue: https://github.com/hashgraph/hedera-services/issues/11751

My PR for addressing this has been blocked for higher priority work and requires coordination across teams. 

PR: https://github.com/hashgraph/hedera-services/pull/12112

Reply:
I can modify the implementation to check for size of the list and query appropriately.